staging/src/k8s.io/apiserver/pkg/util/webhook/client.go

```diff
@@ -164,7 +165,7 @@ func (cm *ClientManager) HookClient(cc ClientConfig) (*rest.RESTClient, error) {
 		}
 		cfg.Dial = func(ctx context.Context, network, addr string) (net.Conn, error) {
 			if addr == host {
-				u, err := cm.serviceResolver.ResolveEndpoint(cc.Service.Namespace, cc.Service.Name)
+				u, err := cm.serviceResolver.ResolveEndpoint(cc.Service.Namespace, cc.Service.Name, cc.Service.Port)
```

Comment:
this is a reasonable spot at which to treat `cc.Service.Port == 0` as `443`, since ClientConfig/ClientConfigService can be constructed/used directly